src/app/util/ember-compatibility-functions.cpp

```diff
@@ -216,218 +216,221 @@ CHIP_ERROR ReadSingleClusterData(FabricIndex aAccessingFabricIndex, const Concre
+            {
+            case ZCL_NO_DATA_ATTRIBUTE_TYPE: // No data
+                ReturnErrorOnFailure(writer->PutNull(tag));
+                break;
+            case ZCL_BOOLEAN_ATTRIBUTE_TYPE: // Boolean
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<bool>(*writer, isNullable));
+                break;
+            case ZCL_INT8U_ATTRIBUTE_TYPE: // Unsigned 8-bit integer
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint8_t>(*writer, isNullable));
+                break;
+            case ZCL_INT16U_ATTRIBUTE_TYPE: // Unsigned 16-bit integer
             {
-                if (isNullable)
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint16_t>(*writer, isNullable));
+                break;
+            }
+            case ZCL_INT32U_ATTRIBUTE_TYPE: // Unsigned 32-bit integer
+            {
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint32_t>(*writer, isNullable));
+                break;
+            }
+            case ZCL_INT64U_ATTRIBUTE_TYPE: // Unsigned 64-bit integer
+            {
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint64_t>(*writer, isNullable));
+                break;
+            }
+            case ZCL_INT8S_ATTRIBUTE_TYPE: // Signed 8-bit integer
+            {
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<int8_t>(*writer, isNullable));
+                break;
+            }
+            case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
+            {
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<int16_t>(*writer, isNullable));
+                break;
+            }
+            case ZCL_INT32S_ATTRIBUTE_TYPE: // Signed 32-bit integer
+            {
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<int32_t>(*writer, isNullable));
+                break;
+            }
+            case ZCL_INT64S_ATTRIBUTE_TYPE: // Signed 64-bit integer
+            {
+                ReturnErrorOnFailure(attributeBufferToNumericTlvData<int64_t>(*writer, isNullable));
+                break;
+            }
+            case ZCL_CHAR_STRING_ATTRIBUTE_TYPE: // Char string
+            {
+                char * actualData  = reinterpret_cast<char *>(attributeData + 1);
+                uint8_t dataLength = attributeData[0];
+                if (dataLength == 0xFF)
                 {
-                    ReturnErrorOnFailure(writer->PutNull(tag));
+                    if (isNullable)
+                    {
+                        ReturnErrorOnFailure(writer->PutNull(tag));
+                    }
+                    else
+                    {
+                        return CHIP_ERROR_INCORRECT_STATE;
+                    }
                 }
                 else
                 {
-                    return CHIP_ERROR_INCORRECT_STATE;
+                    ReturnErrorOnFailure(writer->PutString(tag, actualData, dataLength));
```

Comment:
BTW don't be fooled by diffs like this, I didn't change the error code here, it's just the diff being confused with where it is in the code.